### PR TITLE
Improve the survey and telemetry

### PR DESCRIPTION
**Pull request by kceiw**
## Description

This includes the improvements in showing the survey message and collecting telemetry.
1. Add a new cmdlet `Open-AzSurveyLink` to open the survey in the default browser.
2. Show the survey message three times, in the intervals of 3, 2, 5 days.  The first time is shown after 3 days usage, the second 2 days afterwards, and the third 5 days afterwards.
4. Collect the command name in CommandHistory event for non-supported commands. For support commands, we collect the command name and parameter names.
5. Fix "CommandId" in the telemetry event to correlate the CommandHistory and events proceeding to it.

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/919
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: tools/Az.Tools.Predictor/Az.Tools.Predictor/Commands/OpenAzSurveyLink.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        protected override void ProcessRecord()
+        {
+            var profileSettings = Settings.GetProfileSettings();
+            var surveyId = profileSettings?.SurveyId?.ToString(CultureInfo.InvariantCulture) ?? "000000";
+
+            var link = string.Format(OpenAzSurveyLink._SurveyLinkFormat, surveyId, CultureInfo.InvariantCulture);
+
+            Console.WriteLine($"Opening survey {link}");
+
+            // TODO [mahuang] Ouput the link for user to copy/paste in case it's not open in the default browser.
```
Comment (mirdaki) [inline tools/Az.Tools.Predictor/Az.Tools.Predictor/Commands/OpenAzSurveyLink.cs:50]: Are you able to do this now or is this a later thing?

File: tools/Az.Tools.Predictor/Az.Tools.Predictor/Utilities/JsonUtilities.cs [hunk 2]
```diff
@@ -29,16 +28,20 @@ internal static class JsonUtilities
         /// The default serialization options:
         /// 1. Use camel case in the naming.
         /// 2. Use string instead of number for enums.
+        /// 3. Use the string values (camel case) for enum.
+        /// 4. Use the string values for the type <see cref="System.Version"/> in the properties and <see cref="System.Collections.Generic.IDictionary&lt;TKey, TValue>"/> keys.
+        /// 3. Skip the property if the value is null.
```
Comment (mirdaki) [inline tools/Az.Tools.Predictor/Az.Tools.Predictor/Utilities/JsonUtilities.cs:33]: Nit: 5

**Review by mirdaki — APPROVED** [review]
Just nits

File: tools/Az.Tools.Predictor/Az.Tools.Predictor/Commands/OpenAzSurvey.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                    var fileContent = JsonSerializer.Serialize<Settings>(profileSettings, new JsonSerializerOptions(JsonUtilities.DefaultSerializerOptions)
+                        {
+                            WriteIndented = true,
+                        });
+                    var profileSettingFilePath = Settings.GetProfileSettingsFilePath();
+                    File.WriteAllText(profileSettingFilePath, fileContent, Encoding.UTF8);
+                }
+                catch
+                {
+                    // Ignore all exceptions.
```
Comment (dcaro) [inline tools/Az.Tools.Predictor/Az.Tools.Predictor/Commands/OpenAzSurvey.cs:72]: @kceiw / @VeryEarly  Should we display here a message with the link if we cannot open the default browser?
For example: "The default browser could not be opened, please copy the following link to a browser to give us yourfeedback"

Comment (dcaro) [reply tools/Az.Tools.Predictor/Az.Tools.Predictor/Commands/OpenAzSurvey.cs:72]: Just chatted with @kceiw  ignore this comment, displaying the survey link on the output of the cmdlet is good.

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**